src/OpenColorIO/fileformats/FormatMetadata.h

```diff
@@ -0,0 +1,148 @@
+/*
+Copyright (c) 2018 Autodesk Inc., et al.
+All Rights Reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are
+met:
+* Redistributions of source code must retain the above copyright
+  notice, this list of conditions and the following disclaimer.
+* Redistributions in binary form must reproduce the above copyright
+  notice, this list of conditions and the following disclaimer in the
+  documentation and/or other materials provided with the distribution.
+* Neither the name of Sony Pictures Imageworks nor the names of its
+  contributors may be used to endorse or promote products derived from
+  this software without specific prior written permission.
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+"AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+#ifndef INCLUDED_OCIO_OPS_METADATA_H
+#define INCLUDED_OCIO_OPS_METADATA_H
+
+#include <list>
+#include <string>
+#include <vector>
+
+#include <OpenColorIO/OpenColorIO.h>
+
+#include "PrivateTypes.h"
+
+OCIO_NAMESPACE_ENTER
+{
+
+static constexpr const char * METADATA_INFO = "Info";
```

Comment:
Should the metadata keys have consistent casing? Or are Info and Description special cases (CLF/CTF specific)?

i.e.
Info, Name, Id, Description
OR
info, name, id, description

Reply:
Michael, fair question, but we are following the CLF spec, those are the spellings used in S-2014-006.  There is a logic to it though since the lower-case ones (id, name) are XML attributes whereas the upper-case ones (Info, Description) are XML elements.

Reply:
Ok, that makes sense, and leads to my second question. Is it confusing to lump both under the same `METADATA_*` naming scheme given that they are not intended to be used in the same way. Maybe it doesn't matter since they are not publicly facing, but outside of reading the explanation in FormatMetadata.h, it could be unclear that they carry a more specific intended use than just as metadata keys.

Reply:
I think it is more clear where they are used since ID and NAME show up in getAttribute type calls and INFO and DESCRIPTION show up in getElement type calls.  But I agree this header could use more explanation.  

Bernard, what I would suggest is the following:
// INFO and DESCRIPTION  are CLF XML elements described in S-2014-006.
... the info and desc declarations
// NAME and ID are CLF XML attributes described in S-2014-006.
... the name and id declarations
// ROOT is simply a placeholder ... (4 line comment from below)
... the root declaration

Reply:
That would help clarify. Thanks!

Reply:
done. Will be part of an upcoming commit.